File: TheSky/JavaScript/LZ/scenes/MainScene/Shadow/TrainShadowLayer.js
var TrainShadowLayer = cc.Layer.extend({
	
	/**
	 * @param type 0炼影 1影纹 2影箱
	 */
	ctor:function(type){
		this._super();
		this.type = type || 0;
		this.initLayer();
	},
	initLayer:function(){
		cc.BuilderReader.registerController("TrainShadowViewOwner", {
		});
		this.node = cc.BuilderReader.load(ccbi_res.TrainShadowView_ccbi, this);
		if (this.node != null) {
			this.addChild(this.node);
		}
		this.updataShadowLayer();
	},
	setSpriteFrame:function(sender, bool){
		if (bool) {
			sender.setNormalImage(new cc.Sprite("#btn_fenye_1.png"));
			sender.setSelectedImage(new cc.Sprite("#btn_fenye_1.png"));
		} else {
			sender.setNormalImage(new cc.Sprite("#btn_fenye_0.png"));
			sender.setSelectedImage(new cc.Sprite("#btn_fenye_1.png"));
		}
	},
	updataShadowLayer:function(){
		this.TrainShadowContentLayer.removeAllChildren(true);
		if (this.type == 0) {
			var layer = new TrainView();
			this.TrainShadowContentLayer.addChild(layer);
			this.setSpriteFrame(this.TrainViewBtn, true);
			this.setSpriteFrame(this.TrainWaveBtn, false);
			this.setSpriteFrame(this.TrainBoxBtn, false);
		} else if (this.type == 1) {
			this.TrainShadowContentLayer.addChild(new ShadowWave());
			this.setSpriteFrame(this.TrainViewBtn, false);
			this.setSpriteFrame(this.TrainWaveBtn, true);
			this.setSpriteFrame(this.TrainBoxBtn, false);
		} else if (this.type == 2) {
			this.TrainShadowContentLayer.addChild(new ShadowBox());
			this.setSpriteFrame(this.TrainViewBtn, false);
			this.setSpriteFrame(this.TrainWaveBtn, false);
			this.setSpriteFrame(this.TrainBoxBtn, true);
		}
	},
	TrainShadowAction:function(){
		this.type = 0;
		this.updataShadowLayer();
	},
	shadowWaveAction:function(){
		this.type = 1;
		this.updataShadowLayer();
	},
	shadowBoxAction:function(){
		this.type = 2;
		this.updataShadowLayer();
	},
	changeShadowAction:function(){
		postNotifcation(NOTI.GOTO_TEAM, {page:0, type:1});
	},
	onEnter:function(){
		this._super();
	},
	onExit:function(){
		this._super();
	},
});

File: TheSky/lua/op/TrainShadow/TrainShadowLayer.lua
local _layer

local _pageNumb

local TrainViewBtn
local TrainWaveBtn
local TrainBoxBtn
local LogueTitleLayer
local TrainShadowContentLayer

TrainShadowViewOwner = TrainShadowViewOwner or {}
ccb["TrainShadowViewOwner"] = TrainShadowViewOwner

local function setSpriteFrame( sender,bool )
    if bool then
        sender:setNormalSpriteFrame(CCSpriteFrameCache:sharedSpriteFrameCache():spriteFrameByName("tab_btn_1.png"))
        sender:setSelectedSpriteFrame(CCSpriteFrameCache:sharedSpriteFrameCache():spriteFrameByName("tab_btn_1.png"))
    else
        sender:setNormalSpriteFrame(CCSpriteFrameCache:sharedSpriteFrameCache():spriteFrameByName("tab_btn_0.png"))
        sender:setSelectedSpriteFrame(CCSpriteFrameCache:sharedSpriteFrameCache():spriteFrameByName("tab_btn_1.png"))
    end
end

local function TrainShadowAction( tag,sender )
    TrainShadowContentLayer:removeAllChildrenWithCleanup(true)
    TrainShadowContentLayer:addChild(createTrainView())
    setSpriteFrame(TrainViewBtn,true)
    setSpriteFrame(TrainWaveBtn,false)
    setSpriteFrame(TrainBoxBtn,false)
end

local function shadowWaveAction(  )
    Global:instance():TDGAonEventAndEventData("prop")
    TrainShadowContentLayer:removeAllChildrenWithCleanup(true)
    TrainShadowContentLayer:addChild(createShadowWaveLayer())
    setSpriteFrame(TrainViewBtn,false)
    setSpriteFrame(TrainWaveBtn,true)
    setSpriteFrame(TrainBoxBtn,false)
end

local function shadowBoxAction(  )
    Global:instance():TDGAonEventAndEventData("gift")
    TrainShadowContentLayer:removeAllChildrenWithCleanup(true)
    TrainShadowContentLayer:addChild(createShadowBoxLayer())
    setSpriteFrame(TrainViewBtn,false)
    setSpriteFrame(TrainWaveBtn,false)
    setSpriteFrame(TrainBoxBtn,true)
end

local function changeShadowAction(  )
    if getMainLayer() then
        getMainLayer():gotoTeam()
    end
end

TrainShadowViewOwner["TrainShadowAction"] = TrainShadowAction
TrainShadowViewOwner["shadowWaveAction"] = shadowWaveAction
TrainShadowViewOwner["shadowBoxAction"] = shadowBoxAction
TrainShadowViewOwner["changeShadowAction"] = changeShadowAction

local function gotoPageByNumb( pageNumb )
    TrainShadowContentLayer:removeAllChildrenWithCleanup(true)
    if pageNumb == 0 then
        TrainShadowContentLayer:addChild(createTrainView())
        setSpriteFrame(TrainViewBtn,true)
        setSpriteFrame(TrainWaveBtn,false)
        setSpriteFrame(TrainBoxBtn,false)
    elseif pageNumb == 1 then
        TrainShadowContentLayer:addChild(createShadowWaveLayer())
        setSpriteFrame(TrainViewBtn,false)
        setSpriteFrame(TrainWaveBtn,true)
        setSpriteFrame(TrainBoxBtn,false)
    elseif pageNumb == 2 then
        TrainShadowContentLayer:addChild(createShadowBoxLayer())
        setSpriteFrame(TrainViewBtn,false)
        setSpriteFrame(TrainWaveBtn,false)
        setSpriteFrame(TrainBoxBtn,true)
    end

end
-- private方法如果没有上下调用关系可以写在createLayer外面
local function _init()
    -- get layer
    local  proxy = CCBProxy:create()
    local  node  = CCBReaderLoad("ccbResources/TrainShadowView.ccbi",proxy, true,"TrainShadowViewOwner")
    _layer = tolua.cast(node,"CCLayer")
    TrainShadowContentLayer = tolua.cast(TrainShadowViewOwner["TrainShadowContentLayer"],"CCLayer")

    -- _updatePackageCount()

    TrainViewBtn = tolua.cast(TrainShadowViewOwner["TrainViewBtn"],"CCMenuItemImage")
    TrainWaveBtn = tolua.cast(TrainShadowViewOwner["TrainWaveBtn"],"CCMenuItemImage")
    
    TrainBoxBtn = tolua.cast(TrainShadowViewOwner["TrainBoxBtn"],"CCMenuItemImage")
    gotoPageByNumb( _pageNumb )
end

local function setMenuPriority()
    local menu1 = tolua.cast(TrainShadowViewOwner["myCCMenu"], "CCMenu")
    menu1:setHandlerPriority(-129)
end

function getTrainShadowLayer()
    return _layer
end
-- 0 开始
function createTrainShadowView(pageNumb)
    _pageNumb = pageNumb and pageNumb or 0
    -- if pageNumb then
    --     _pageNumb = pageNumb
    -- end
    _init()

    local function _onEnter()
        local seq = CCSequence:createWithTwoActions(CCDelayTime:create(0.5), CCCallFunc:create(setMenuPriority))
        _layer:runAction(seq)
    end

    local function _onExit()
        _pageNumb = nil
    end

    --onEnter onExit
    local function _layerEventHandler(eventType)
        if eventType == "enter" then
            if _onEnter then _onEnter() end
        elseif eventType == "exit" then
            if _onExit then _onExit() end
        end
    end


    _layer:registerScriptHandler(_layerEventHandler)

    return _layer
end